inc/saiswitch.h

```diff
@@ -1995,6 +2237,62 @@ typedef void (*sai_switch_state_change_notification_fn)(
         _In_ sai_object_id_t switch_id,
         _In_ sai_switch_oper_status_t switch_oper_status);
 
+/**
+ * @brief Platform specific device register read access
+ *
+ * This API provides platform adaption in 2 ways
+ *
+ * 1# Provides platform adaption functionality to access device
+ * registers from driver. This is mandatory to pass as attribute to
+ * sai_create_switch when driver implementation does not support register access
+ * by device file system directly.
+ * 2# Provides device access API from ASIC SDK.
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] platform_context Platform context information.
+ * @param[in] switch_id Switch Id
+ * @param[in] device_addr Device address(PHY/lane/port MDIO address)
+ * @param[in] start_reg_addr Starting register address to read
+ * @param[in] number_of_registers Number of consecutive registers to read
+ * @param[out] reg_val Register read values
+ */
+typedef sai_status_t (*sai_switch_register_read_fn)(
+        _In_ const void *platform_context,
```

Comment:
why platform context is extra parameter since you have switch id already for identifying switch itself

Reply:
also please avoid void* abitrary data, since it will be impossible to guess whats there and how to pass this over RPC (which is currently done in sonic), please consider using switch attributes, without any pointers etc.

Reply:
also change this platform_context to u64